source/base/data_out_base.cc

```diff
@@ -7715,13 +7718,15 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
       std::stringstream ss;
       DataOutBase::write_vtu_header(ss, vtk_flags);
       header_size = ss.str().size();
-      // Write the header on rank 0 and automatically move the
-      // shared file pointer to the location after header;
-      ierr = MPI_File_write_shared(
-        fh, ss.str().c_str(), header_size, MPI_CHAR, MPI_STATUS_IGNORE);
+      // Write the header on rank 0 at the starting of a file, i.e., offset 0.
+      ierr = MPI_File_write_at(
+        fh, 0, ss.str().c_str(), header_size, MPI_CHAR, MPI_STATUS_IGNORE);
       AssertThrowMPI(ierr);
     }
 
+  ierr = MPI_Bcast(&header_size, 1, MPI_UNSIGNED, 0, comm);
+  AssertThrowMPI(ierr);
```

Comment:
Can you use the following instead?
```suggestion
  header_size = Utilities::MPI::broadcast(header_size, comm, 0);
```

Reply:
I think that function has an unfortunate design with using the return value and no real advantage for a single int over Bcast. But we can make the change if you insist.

Reply:
I find it easier to read with the return value, but no, I don't insist.

Reply:
(Also substantially less easy to make mistakes when someone changes the code because, for example, it automatically detects the type of the object being broadcasted.)